nucypher/blockchain/eth/sol/source/contracts/StakingEscrow.sol

```diff
@@ -910,6 +1039,16 @@ contract StakingEscrow is Issuer, IERC900History {
         addSnapshot(info, - int256(_value));
         token.safeTransfer(msg.sender, _value);
         emit Withdrawn(msg.sender, _value);
+
+        // unbond worker if staker withdraws last portion of NU
+        if (info.value == 0 &&
+            info.nextCommittedPeriod == 0 &&
+            info.worker != address(0))
+        {
+            stakerFromWorker[info.worker] = address(0);
+            info.worker = address(0);
```

Comment:
At first it was there, but I removed to minimize changes because we have case when worker is `address(0)` in `bondWorker` to bypass check for `workerStartPeriod`